### DISCUSSION TITLE
[CRITICAL] Failed to load grains defined in grain file lvm.lvm in function <function lvm at 0x7f4ca258c730>

### DISCUSSION THREAD

**Discussion by byronsims** [Security-Onion-Solutions/securityonion#3892] [Unsupported Versions] [answered]
When I run so-status I get the following.  Is there a way for me to fix this without resinstalling?  This is on a sensor node.

[CRITICAL] Failed to load grains defined in grain file lvm.lvm in function <function lvm at 0x7f4ca258c730>, error:
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 853, in grains
    ret = funcs[key](**kwargs)
  File "/usr/lib/python3.6/site-packages/salt/grains/lvm.py", line 27, in lvm
    return _linux_lvm()
  File "/usr/lib/python3.6/site-packages/salt/grains/lvm.py", line 44, in _linux_lvm
    "{} lvs -o lv_name --noheadings {}".format(cmd, vg)
  File "/usr/lib/python3.6/site-packages/salt/modules/cmdmod.py", line 893, in _run_quiet
    success_retcodes=success_retcodes,
  File "/usr/lib/python3.6/site-packages/salt/modules/cmdmod.py", line 663, in _run
    cmd = salt.utils.args.shlex_split(cmd)
  File "/usr/lib/python3.6/site-packages/salt/utils/args.py", line 318, in shlex_split
    shlex.split(salt.utils.stringutils.to_str(s), **kwargs)
  File "/usr/lib64/python3.6/shlex.py", line 305, in split
    return list(lex)
  File "/usr/lib64/python3.6/shlex.py", line 295, in __next__
    token = self.get_token()
  File "/usr/lib64/python3.6/shlex.py", line 105, in get_token
    raw = self.read_token()
  File "/usr/lib64/python3.6/shlex.py", line 187, in read_token
    raise ValueError("No closing quotation")
ValueError: No closing quotation

Checking Docker status

    Docker ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    

Checking container statuses

    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-nginx ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-sensoroni ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-steno ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-backend ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-strelka-coordinator -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-filestream --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-frontend ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-gatekeeper --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-manager ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-suricata ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-telegraf ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-wazuh ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-zeek ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

**Comment by jertel** [comment]
Sounds like a corrupt grain file. Check the files in the `/etc/salt` directory for values that look incomplete. Perhaps you entered a quotation character somewhere that doesn't belong.

**Comment by byronsims** [comment]
I do not see anything different between the files on this sensor node compared to my other sensor node.

-Byron

**Reply by byronsims** [reply]
I am still getting this on one of my sensor nodes.  Any other ideas on what I can do to try to fix it?

-Byron

**Comment by byronsims** [comment]
I never did. I’ve been running it in this state and everything seems to continue to work. Sent from my iPhoneOn Nov 7, 2022, at 2:12 PM, elhijo ***@***.***> wrote:﻿
Hi,
Have you found a solution with this issue ?
I have the same issue with one of my nodes.
Thanks

**Comment by elhijo** [comment, marked as answer]
I've found where the issue was yesterday right after posting this message.
Running lvdisplay, I've found some drives marked as unknown, removing them cleared this error.
Hope that can help.
